### DISCUSSION TITLE
forever start nextjs on custom port

### DISCUSSION THREAD

**Discussion by ivankopcik** [vercel/next.js#15193] [Help]
Hello guys,

what is the proprer way to start **production** (builded) version of nextjs project on different ports via forever? We are starting domains on different ports.

I already searched whole internet..., tried:

`PORT=5605` in `.env.production` and bunch of other solutions

Thank you

**Comment by cristiand391** [comment]
You can specify the port on which your app will start using the `-p` flag.
If you wanna set different ports in production and development, you can set them in their respective `.env` file and then read the `PORT` env var using `$`.

Example:
```
// '.env.development' file
PORT=3000

// '.env.production' file
PORT=5605

// 'package.json' file
...
"scripts": {
  ...
  "start": "next start -p $PORT"
}
...
```

**Reply by pandaa880** [reply]
I tried this and I am getting `Unhandled promise rejection warning`. I don't understand what am I doing wrong. I tried this with `.env` `.env.development` `.env.production`. I think `$PORT` is not meant for `package.json`. From what I understood from searching and reading up, there is no way to load PORT from `.env` file at the moment. You have to pass the `-p 4011` explicitly while running the command or set it in **package.json** or have a custom server.

**Reply by cristiand391** [reply]
Which version of Next.js are you using?

**Reply by pandaa880** [reply]
I am using `9.5.3`.

**Reply by cristiand391** [reply]
I think I was working with a dockerized app when using that workflow, sorry.

> I think `$PORT` is not meant for `package.json`.

Whenever you run the `dev`, `build`, or `start` script, it is executed in your terminal so it seach for PORT.

You can check that by setting them inside your package.json as follows:
```
...
"scripts": {
  "dev": "PORT=8080 && next dev -p $PORT",   // started server  on http://localhost:8080
  "build": "next build",
  "start": "PORT=5000 && next start -p $PORT" // started server on http://localhost:5000
...
```

**Reply by pandaa880** [reply]
Yes, but for this I have to edit the `package.json` everytime I want to change the port. What I wanted was to read **PORT** from `.env` like `create-react-app`.

If I do it like this 
```js
...
"scripts": {
  "dev": "PORT=8080 && next dev -p $PORT",   // started server  on http://localhost:8080
  "build": "next build",
  "start": "PORT=5000 && next start -p $PORT" // started server on http://localhost:5000
...
```

then I might as well specify the port number directly on $PORT `next dev -p 3022`.

Do you have any resource for achieving this with **Docker** ?

**Reply by cristiand391** [reply]
[Docker compose docs](https://docs.docker.com/compose/environment-variables/#substitute-environment-variables-in-compose-files)

**Reply by ivankopcik** [reply]
Exactly, this solution is not optimal, because when we run multiple applications on server we need to change `.env.local` and `package.json` which makes merge confilcts. 

Ideal solution is to edit `.env.local` and app should take $PORT from .env
